web-app/js/portal/data/MetadataRecordFetcher.js

```diff
@@ -34,10 +34,11 @@ Portal.data.MetadataRecordFetcher = Ext.extend(Ext.util.Observable, {
             var store = new Portal.data.MetadataRecordStore();
             store.loadData(response.responseXML);
             var metadataRecord = store.getAt(0);
-            var dataCollection = Portal.data.DataCollection.fromMetadataRecord(metadataRecord);
 
             if (metadataRecord) {
+                var dataCollection = Portal.data.DataCollection.fromMetadataRecord(metadataRecord);
                 _this.dataCollectionStore.add(dataCollection);
+
```

Comment:
It's not about how long the block is, it's to separate code that's doing two logically separate operations. The first two lines work together to load the collection and add it to the store (which are data model operations), and the last line sets some UI state.